### PR TITLE
Fix Sanity failures in stable-2.4

### LINKED ISSUE #672: Community package requirements: sanity tests
(Note: This issue was filed in a semi-automated fashion. Let me know if you see errors in this issue.)

As per the [Ansible community package inclusion requirements][ci-testing], collections must pass `ansible-test sanity` tests. Version `2.4.0` of `kubernetes.core`, corresponding to the `2.4.0` tag in this repo, fails one or more of the required sanity tests.

Please see the errors below and address them. If these issues aren't addressed within a reasonable time period, the collection may be subject to [removal from Ansible][removal].

Thank you for your efforts and for being part of the Ansible package! We appreciate it.

---

## Sanity tests

The following tests were run using `ansible-test` version `2.16.1`:

- ansible-doc
- compile
- validate-modules
- yamllint

Note that this is only a subset of the required sanity tests. Please make sure you run them in all in your CI.

### Results

> **💡 NOTE:**
>
> Check the `[explain]` links below for more information about each test and how to fix failures.
> See [Sanity Tests: Ignores](https://docs.ansible.com/ansible/latest/dev_guide/testing/sanity/ignores.html) in the dev guide if, after reading the test-specific documentation, you still believe an error is a false positive.

The test `ansible-test sanity --test validate-modules` [[explain](https://docs.ansible.com/ansible-core/2.16/dev_guide/testing/sanity/validate-modules.html)] failed with 25 errors:

``` text
plugins/modules/helm_info.py:0:0: doc-default-does-not-match-spec: Argument 'release_state' in argument_spec defines default as ([]) but documentation defines default as (None)
plugins/modules/helm_template.py:0:0: doc-default-does-not-match-spec: Argument 'show_only' in argument_spec defines default as ([]) but documentation defines default as (None)
plugins/modules/k8s.py:0:0: parameter-type-not-in-doc: Argument 'resource_definition' in argument_spec defines type as <function list_dict_str at 0x7f25ee49c180> but documentation doesn't define type
plugins/modules/k8s.py:0:0: return-syntax-error: RETURN.result.contains.error.contains: required key not provided @ data['result']['contains']['error']['contains']. Got None
plugins/modules/k8s.py:0:0: return-syntax-error: RETURN.result.contains.metadata.contains: required key not provided @ data['result']['contains']['metadata']['contains']. Got None
plugins/modules/k8s.py:0:0: return-syntax-error: RETURN.result.contains.spec.contains: required key not provided @ data['result']['contains']['spec']['contains']. Got None
plugins/modules/k8s.py:0:0: return-syntax-error: RETURN.result.contains.status.contains: required key not provided @ data['result']['contains']['status']['contains']. Got None
plugins/modules/k8s_drain.py:0:0: doc-default-does-not-match-spec: Argument 'delete_options' in argument_spec defines default as ({}) but documentation defines default as (None)
plugins/modules/k8s_info.py:0:0: doc-default-does-not-match-spec: Argument 'field_selectors' in argument_spec defines default as ([]) but documentation defines default as (None)
plugins/modules/k8s_info.py:0:0: doc-default-does-not-match-spec: Argument 'label_selectors' in argument_spec defines default as ([]) but documentation defines default as (None)
plugins/modules/k8s_log.py:0:0: doc-default-does-not-match-spec: Argument 'label_selectors' in argument_spec defines default as ([]) but documentation defines default as (None)
plugins/modules/k8s_rollback.py:0:0: doc-default-does-not-match-spec: Argument 'field_selectors' in argument_spec defines default as ([]) but documentation defines default as (None)
plugins/modules/k8s_rollback.py:0:0: doc-default-does-not-match-spec: Argument 'label_selectors' in argument_spec defines default as ([]) but documentation defines default as (None)
plugins/modules/k8s_scale.py:0:0: doc-default-does-not-match-spec: Argument 'label_selectors' in argument_spec defines default as ([]) but documentation defines default as (None)
plugins/modules/k8s_scale.py:0:0: parameter-type-not-in-doc: Argument 'resource_definition' in argument_spec defines type as <function list_dict_str at 0x7f25ee49c180> but documentation doesn't define type
plugins/modules/k8s_scale.py:0:0: return-syntax-error: RETURN.result.contains.metadata.contains: required key not provided @ data['result']['contains']['metadata']['contains']. Got None
plugins/modules/k8s_scale.py:0:0: return-syntax-error: RETURN.result.contains.spec.contains: required key not provided @ data['result']['contains']['spec']['contains']. Got None
plugins/modules/k8s_scale.py:0:0: return-syntax-error: RETURN.result.contains.status.contains: required key not provided @ data['result']['contains']['status']['contains']. Got None
plugins/modules/k8s_service.py:0:0: parameter-type-not-in-doc: Argument 'resource_definition' in argument_spec defines type as <function list_dict_str at 0x7f25ee49c180> but documentation doesn't define type
plugins/modules/k8s_service.py:0:0: return-syntax-error: RETURN.result.contains.metadata.contains: required key not provided @ data['result']['contains']['metadata']['contains']. Got None
plugins/modules/k8s_service.py:0:0: return-syntax-error: RETURN.result.contains.spec.contains: required key not provided @ data['result']['contains']['spec']['contains']. Got None
plugins/modules/k8s_service.py:0:0: return-syntax-error: RETURN.result.contains.status.contains: required key not provided @ data['result']['contains']['status']['contains']. Got None
plugins/modules/k8s_taint.py:0:0: return-syntax-error: RETURN.result.contains.metadata.contains: required key not provided @ data['result']['contains']['metadata']['contains']. Got None
plugins/modules/k8s_taint.py:0:0: return-syntax-error: RETURN.result.contains.spec.contains: required key not provided @ data['result']['contains']['spec']['contains']. Got None
plugins/modules/k8s_taint.py:0:0: return-syntax-error: RETURN.result.contains.status.contains: required key not provided @ data['result']['contains']['status']['contains']. Got None
```

The test `ansible-test sanity --test yamllint` [[explain](https://docs.ansible.com/ansible-core/2.16/dev_guide/testing/sanity/yamllint.html)] failed with 9 errors:

``` text
tests/integration/targets/helm/files/appversionless-chart-v2/templates/configmap.yaml:6:13: unparsable-with-libyaml: while constructing a mapping - found unhashable key
tests/integration/targets/helm/files/appversionless-chart/templates/configmap.yaml:6:13: unparsable-with-libyaml: while constructing a mapping - found unhashable key
tests/integration/targets/helm/files/test-chart-v2/templates/configmap.yaml:6:13: unparsable-with-libyaml: while constructing a mapping - found unhashable key
tests/integration/targets/helm/files/test-chart/templates/configmap.yaml:6:13: unparsable-with-libyaml: while constructing a mapping - found unhashable key
tests/integration/targets/helm_diff/files/test-chart/templates/configmap.yaml:6:13: unparsable-with-libyaml: while constructing a mapping - found unhashable key
tests/integration/targets/k8s_scale/files/deployment.yaml:26:1: unparsable-with-libyaml: expected a single document in the stream - but found another document
tests/unit/module_utils/fixtures/definitions.yml:6:1: unparsable-with-libyaml: expected a single document in the stream - but found another document
tests/unit/module_utils/fixtures/deployments.yml:25:1: unparsable-with-libyaml: expected a single document in the stream - but found another document
tests/unit/module_utils/fixtures/pods.yml:18:1: unparsable-with-libyaml: expected a single document in the stream - but found another document
```

[ci-testing]: https://docs.ansible.com/ansible/latest/community/collection_contributors/collection_requirements.html#ci-testing
[repo-mgmt]: https://docs.ansible.com/ansible/latest/community/collection_contributors/collection_requirements.html#repository-management
[removal]: https://github.com/ansible-collections/overview/blob/main/removal_from_ansible.rst

### PR DESCRIPTION
**Pull request by GomathiselviS**
##### SUMMARY

Fixes #672 

- Added ignore-2.16.txt
- Fixed issues with module documentation.
- Add github workflows (Sanity tests are removed from Zuul. They are run by GHA)

##### ISSUE TYPE

- Bugfix Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by GomathiselviS** [comment]
recheck

File: .github/workflows/integration-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  integration:
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    needs:
+      - splitter
+    if: ${{ needs.splitter.outputs.test_targets != '' }}
+    env:
+      source: "./source"
+      cloud_common: "./cloudcommon"
+      ansible_posix: "./ansible_posix"
```
Comment (abikouo) [inline .github/workflows/integration-tests.yaml:52]: it should be nice to define all dependent collections into a `tests/integration/requirements.yml` file as we did for others collections

**Review by abikouo — APPROVED** [review]

**Comment by abikouo** [comment]
@GomathiselviS It seems like you need to set the Zuul galaxy importer job as non-voting as we did for others collections

File: .github/workflows/integration-tests.yaml (hunk 1 above)
Comment (GomathiselviS) [reply .github/workflows/integration-tests.yaml:52]: I can do it as part of a separate PR. This PR is to fix the sanity issues.

**Review by alinabuzachis — APPROVED** [review]

File: plugins/filter/k8s.py [hunk 2]
```diff
@@ -1,12 +1,50 @@
 # Copyright (c) 2017 Ansible Project
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
-
+# SPDX-License-Identifier: GPL-3.0-or-later
```
Comment (alinabuzachis) [inline plugins/filter/k8s.py:3]: Is this something intended?

File: tests/sanity/ignore-2.16.txt [hunk 3]
```diff
@@ -0,0 +1,36 @@
+plugins/module_utils/version.py pylint!skip
+plugins/module_utils/client/discovery.py import-3.6!skip
+plugins/module_utils/client/discovery.py import-3.7!skip
+plugins/module_utils/client/discovery.py import-3.8!skip
+plugins/module_utils/client/discovery.py import-3.9!skip
+plugins/module_utils/client/discovery.py import-3.10!skip
+plugins/module_utils/client/discovery.py import-3.11!skip
```
Comment (gravesm) [inline tests/sanity/ignore-2.16.txt:7]: I believe the only thing needed to get the last sanity tests to pass is to add a line for 3.12 here, as well as for `plugins/module_utils/client/resource.py` and `plugins/module_utils/k8sdynamicclient.py`. We should also do this for the 2.17 ignore file.

**Review by gravesm — APPROVED** [review]

**Pull request merged**